monailabel/tasks/infer/basic_infer.py

```diff
@@ -297,7 +302,11 @@ def __call__(
         latency_pre = time.time() - start
 
         start = time.time()
-        data = self.run_inferer(data, device=device)
+        data = (
+            self.run_inferer(data, device=device)
+            if not self.type == InferType.DETECTION
+            else self.run_detector(data, device=device)
```

Comment:
what it makes different to run infer vs detector.  at the end both should probably run network in eval mode.. is not true?

i thought detector is just another model which outputs some predictions and as part of post processing, we do generate ROI/BoundingBox as the output?  am i missing something here?

Reply:
That's the tricky thing here. 

The detection model defined in the bundle has this:

https://github.com/Project-MONAI/model-zoo/blob/1be315fe132f5071830c07617e91d619868559f5/models/lung_nodule_ct_detection/configs/inference.json#L61

The detector is a wrapper of higher concept of inferer in detection task, it contains the networks, box selectors, anchor_generator, inferer, etc. 

The detector can have different networks. 

That's what I understand from the detection bundle side. 
Maybe @Can-Zhao can explain more of this, she helped me understand the design of "detector". :)

Reply:
To help us understand the detection task:

Here is the difference of Detector and Detection Network in MONAI Core:

Detector: 
https://github.com/Project-MONAI/MONAI/blob/6fa4bce1961d84f89c243f27c093054396ee4557/monai/apps/detection/networks/retinanet_detector.py#L65

Detection Network:
https://github.com/Project-MONAI/MONAI/blob/6fa4bce1961d84f89c243f27c093054396ee4557/monai/apps/detection/networks/retinanet_network.py#L199